documentation/modules/oauth/proc-oauth-authorization-keycloak-example.adoc

```diff
@@ -256,14 +283,38 @@ keytool -keystore /tmp/truststore.p12 -storetype pkcs12 -alias sso -storepass $S
 KAFKA_HOST_PORT=my-cluster-kafka-bootstrap:9093
 STOREPASS=storepass
 
-echo "Q" | openssl s_client -showcerts -connect $KAFKA_HOST_PORT 2>/dev/null | awk ' /BEGIN CERTIFICATE/,/END CERTIFICATE/ { print $0 } ' > /tmp/my-cluster-kafka.crt
+echo "Q" | openssl s_client -showcerts -connect $KAFKA_HOST_PORT 2>/dev/null | awk ' /BEGIN CERTIFICATE/,/END CERTIFICATE/ { print $0 } ' > /tmp/my-cluster-kafka.pem
 ----
 
++
+If the client accesses the Kafka cluster over the ingress or OpenShift route, the Kafka bootstrap hostname is different, and you may need to specify it as an extra argument to the `openssl` command in order to get the certificate containing the proper SNA records:
++
+[source,shell]
+----
+KAFKA_BOOTSTRAP_EXTERNAL=kafka-bootstrap-addr.my-domain.ext
+STOREPASS=storepass
+
+echo "Q" | openssl s_client -showcerts -servername $KAFKA_BOOTSTRAP_EXTERNAL -connect $KAFKA_HOST_PORT 2>/dev/null | awk ' /BEGIN CERTIFICATE/,/END CERTIFICATE/ { print $0 } ' > /tmp/my-cluster-kafka.pem
+----
```

Comment:
This does not make sense because above you assume _`tls` listener port (9093)_. If you wanna make it general, you should do it properly, and cover all the listener types and all the possible situations from the beginning. So you need to start from the beginning, how do they find the bootstrap address in the status section of the Kafka YAML. (i.e. how to set `KAFKA_HOST_PORT` properly in the first place).

Reply:
Let's differentiate between example related docs, and non-example related docs. This example runs in an interactive pod in the same namespace where the Kafka cluster is deployed by Strimzi. Access for the example is not via a Route.
This extra fragment tries to be helpful for people who want to access it from outside. It's not about being consistent, it's not about covering all the possible listener types ... Contecting to Kafka cluster in Strimzi through different listener types should be covered elsewhere in the documentation. I suggest I include here the link to such a chapter.

Reply:
> This example runs in an interactive pod in the same namespace where the Kafka cluster is deployed by Strimzi. Access for the example is not via a Route.

Fine, then remove the stuff which does not belong there. Because the other stuff suggests it is a general guide that works for all configurations.

> This extra fragment tries to be helpful for people who want to access it from outside. It's not about being consistent, it's not about covering all the possible listener types ... Connecting to Kafka cluster in Strimzi through different listener types should be covered elsewhere in the documentation. I suggest I include here the link to such a chapter.

Fine, then do it properly and cover the access from outside in the previous steps and for all listeners.

----

Pick one and do it properly, when you do both, it ultimately doesn't work for anyone.